src/status_im/ui/screens/events.cljs

```diff
@@ -106,21 +106,20 @@
 
 (re-frame/reg-fx
   :http-post
-  (fn [{:keys [action data success-event-creator failure-event-creator]}]
-    (utils/http-post action
-                     data
-                     #(re-frame/dispatch (success-event-creator %))
-                     #(re-frame/dispatch (failure-event-creator %)))))
-
-(defn- http-get [{:keys [url response-validator success-event-creator failure-event-creator]}]
-  (if response-validator
-    (utils/http-get url
-                    response-validator
-                    #(re-frame/dispatch (success-event-creator %))
-                    #(re-frame/dispatch (failure-event-creator %)))
-    (utils/http-get url
-                    #(re-frame/dispatch (success-event-creator %))
-                    #(re-frame/dispatch (failure-event-creator %)))))
+  (fn [{:keys [action data success-event-creator failure-event-creator timeout-ms]}]
+    (let [on-success #(re-frame/dispatch (success-event-creator %))
+          on-error   #(re-frame/dispatch (failure-event-creator %))]
+      (cond
+        timeout-ms (utils/http-post action data on-success on-error timeout-ms)
+        :default   (utils/http-post action data on-success on-error)))))
+
+(defn- http-get [{:keys [url response-validator success-event-creator failure-event-creator timeout-ms]}]
+  (let [on-success #(re-frame/dispatch (success-event-creator %))
+        on-error   #(re-frame/dispatch (success-event-creator %))]
+    (cond
```

Comment:
We might benefit from passing args to `utils/http-get` as a map here.

Reply:
@jeluard I only set the last argument a map which will take `:valid-response?` and `:timeout-ms` as optional data, as I assume `url` `on-success` `on-error` are all required, so not putting inside the map. Does it make sense to you?